jobsdb/jobsdb.go

```diff
@@ -2478,15 +2477,16 @@ func (jd *HandleT) markClearEmptyResult(ds dataSetT, workspace string, stateFilt
 			jd.dsEmptyResultCache[ds][workspace][cVal][cValDefaultFilter] = map[string]cacheEntry{}
 		}
 
-		for _, st := range stateFilters {
-			previous := jd.dsEmptyResultCache[ds][workspace][cVal][pVal][st]
-			if checkAndSet == nil || *checkAndSet == previous.Value {
-				cache := cacheEntry{
-					Value: value,
-					T:     time.Now(),
+		for _, pf := range pvalArr {
```

Comment:
what was the issue here?

Reply:
So there are two places where cache value is changed

When we do a store we change entries in the cache map:
- For every job which we get we store cache for the default cache value(cval_cval) and with (destID_xxxx)

Then whenever we do a get we check `isEmptyResult` we check dynamically based on how the query is being made
- with cval_cval is parameter filters are empty
- with destID_xxxx if parameter filters are passed on

Post this after a successful query if we end up having no jobs, then we would need to update the cache for both cval_cval and destID_xxxx, but the problem is the query might send either valid param filters or empty filters and so we need to account the updation for both the cases

Reply:
no I don't think there is a way to properly test it, I was just trying to make a stronger point of why the current cache implementation needs to be redesigned!